src/Xamarin.Forms.StateSquid.Multi/Platforms/Shared/StateLayoutController.cs

```diff
@@ -170,5 +177,27 @@ View CreateItemView(State state, string customState)
 
             return new Label() { Text = $"Template for {state.ToString()}{customState} not defined." };
         }
+
+        private async Task ChildrenFadeTo(Layout<View> layout, bool animate, bool isHide)
+        {
+            if (animate && layout?.Children?.Count > 0)
+            {
+                int duration = isHide ? 100 : 500 / layout.Children.Count;
+                uint uDuration = (uint)(duration > 1 ? duration : 1);
+
+                foreach (var a in layout.Children)
```

Comment:
Doesn't this make it so that each child is awaited individually and in sequence (meaning the total load time is the sum of the animations)? I guess your calculation for `duration` takes care of this, but I wonder if this shouldn't be changed into a list of tasks that get awaited as a single unit using `Task.WhenAll`.